drivers/soundwire/intel.c

```diff
@@ -1268,15 +1268,12 @@ static struct sdw_master_ops sdw_intel_ops = {
 	.post_bank_switch = intel_post_bank_switch,
 };
 
-static int intel_init(struct sdw_intel *sdw, bool init_master)
+static int intel_init(struct sdw_intel *sdw)
 {
 	/* Initialize shim and controller */
 	intel_link_power_up(sdw);
 	intel_shim_init(sdw);
 
-	if (!init_master)
-		return 0;
-
```

Comment:
@plbossart  it is not always true.  If audio pci device is in D0 or D0i3,  master can preserve its context, so we don't need to reinitialize it again.  And if master is in clock stop state, we can't update config before clearing clock_stop in MCP_CONTROL, so sdw_cdns_init will return config update timeout.

Reply:
hi Pierre,  I make another PR to fix a io issue found by QA on this branch. please 
 help to review it it https://github.com/plbossart/sound/pull/8. Thanks!  
We don't need to revert above change if this PR is used.

Reply:
@RanderWang I merged all your code from plbossart#8 in this PR (with a couple of cleanups)